sdk/core/azure-core/src/http/telemetry_policy.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <http/policy.hpp>
+
+#include <sstream>
+
+using namespace Azure::Core::Http;
+
+std::string const TelemetryPolicy::g_emptyApplicationId;
+
+std::string TelemetryPolicy::BuildTelemetryId(
+    std::string const& componentName,
+    std::string const& componentVersion,
+    std::string const& applicationId)
+{
+  // Spec: https://azure.github.io/azure-sdk/general_azurecore.html#telemetry-policy
+  std::ostringstream telemetryId;
+
+  if (!applicationId.empty())
+  {
+    telemetryId << applicationId.substr(0, 24) << " ";
```

Comment:
The applicationId might have whitespace at the end.  Trim the whitespace.

Reply:
Resolved.